### DISCUSSION TITLE
How can data be stored in S3 and retrieve it via LyraSearch?

### DISCUSSION THREAD

**Discussion by raducretu** [oramasearch/orama#204] [Q&A] [answered]
I would like to keep a data.json in S3 and query the data server-side, via API Gateway or AppSync, can this be done? or a worker for search only... with the persistent data plugin ?

**Comment by ShogunPanda** [comment, marked as answer]
Hello!
If you use Lambda and S3, you can use [nebula](https://github.com/LyraSearch/nebula) to easily deploy lyra on AWS.
Otherwise, you still look on how we do [here](https://github.com/LyraSearch/nebula/tree/main/src/targets/aws-lambda) and tailor your own solution. The idea behind it is to pre-bundle lyra in a self contained JS file and configure it to download a file from S3 via HTTP when deserializing.

Let me know if you have any other questions :)
